File: apps/android/feature/ai/domain/src/main/java/com/scribblefit/feature/ai/domain/engine/AnalysisEngine.kt
package com.scribblefit.feature.ai.domain.engine

import com.scribblefit.feature.ai.domain.model.AnalysisSuggestion
import com.scribblefit.feature.ai.domain.model.AnalysisSummary
import com.scribblefit.feature.ai.domain.model.ExerciseInsight
import com.scribblefit.feature.ai.domain.model.SummaryPeriod

interface AnalysisEngine {
    suspend fun generateSuggestion(context: String): Result<AnalysisSuggestion>
    suspend fun generateSummary(period: SummaryPeriod, workoutData: String): Result<AnalysisSummary>
    suspend fun generateExerciseInsight(exerciseName: String, historyData: String): Result<ExerciseInsight>
}

File: apps/ios/ScribbleFit/ScribbleFit/Core/AI/AnalysisEngine.swift
import Foundation

public protocol AnalysisEngine: Sendable {
    func generateSuggestion(context: String) async throws -> AnalysisSuggestion
    func generateSummary(period: SummaryPeriod, workoutData: String) async throws -> AnalysisSummary
    func generateExerciseInsight(exerciseName: String, historyData: String) async throws -> ExerciseInsight
}
